### PR TITLE
DataVolume bind mount for all agent instances

### PR DESCRIPTION
**Pull request by alena1108**
@cloudnautique @ibuildthecloud

### CODE REVIEW

File: code/iaas/logic/src/main/java/io/cattle/platform/process/instance/InstancePreCreate.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+    @Override
+    public String[] getProcessNames() {
+        return new String[] { "instance.create" };
+    }
+
+    @Override
+    public HandlerResult handle(ProcessState state, ProcessInstance process) {
+        Instance instance = (Instance)state.getResource();
+        if (!InstanceConstants.CONTAINER_LIKE.contains(instance.getKind())) {
```
Comment (alena1108) [inline code/iaas/logic/src/main/java/io/cattle/platform/process/instance/InstancePreCreate.java]: Not 100% sure if we need this limitation... @ibuildthecloud ^^

**Comment by cloudnautique** [comment]
Could we add the -v /var/lib/rancher/etc:/var/lib/rancher/etc:ro to the custom command for registering hosts. we will be directing users to place the ssl cert in that folder.

**Comment by cloudnautique** [comment]
Load balancers are able to activate now with these changes.

**Comment by ibuildthecloud** [comment]
@cloudnautique Do we really need the user to put `-v /var/lib...` in the register command being that that command will just launch the actual rancher-agent container?

**Comment by cloudnautique** [comment]
Yes. The first container talks to rancher server, and the first container wires up everything for the agent.

**Comment by cloudnautique** [comment]
@alena1108, @ibuildthecloud is ok with the bind mount being added to the run command.

**Pull request merged**